arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -181,7 +195,68 @@ def run_timerlat(
                 total_usr_latency[label] = line_list[3]
 
         # Provide the rtla command formatted data as debug output
-        print(output)
+        print(timerlat_output)
+
+        # Example time series output from the tracer (truncated)
+        # <idle>-0      [009] d. 123.769498: #1002 context    irq timer_latency  458 ns
+        #  <...>-625890 [009] .. 123.769499: #1002 context thread timer_latency 666 ns
+        #  <...>-625890 [009] .. 123.769500: #1002 context user-ret timer_latency 499 ns
+        # <idle>-0      [002] d. 123.769528: #1003 context    irq timer_latency  587 ns
+        #  <...>-625883 [002] .. 123.769532: #1003 context thread timer_latency 712 ns
+        #  <...>-625883 [002] .. 123.769534: #1003 context user-ret timer_latency 462 ns
+
+        if params.enable_time_series:
+            timeseries_lines = iter(timeseries_output.splitlines())
+            timeseries_dict = {}
+
+            # The calculation of the offset from uptime to current time certainly means
+            # that our time series is not 100% accurately aligned to the nanosecond, but
+            # the relative times will be accurate. We'll accept this as good enough.
+            uptime_offset = time.time() - time.monotonic()
+
+            if all(False for _ in timeseries_lines):
```

Comment:
This ought to "eat" the first element in your iterator so that the subsequent `for` loop won't see it. Is that what you intended, or is this a bug?

Why not just

```python
timeseries_lines = timeseries_output.splitlines()
if len(timeseries_lines) == 0:
    print(...)
for line in timeseries_lines:
    ...
```

Or

```python
timeseries_lines = timeseries_output.splitlines()
for line in timeseries_lines:
    ...
else:
    print("No results ...")
```

Reply:
Oh ... you're probably right. The `else` triggers only if there's no `break`, which we don't want here. Sigh. Well, it seemed like a good idea at the time. 😦

Reply:
Since we're simply testing for the presence of output, I think we should be looking at `timeseries_output` rather than the iterator that we created for it.